### DISCUSSION TITLE
What's the origin of Sphere Mapping Texture UV?

### DISCUSSION THREAD

**Discussion by WangHoi** [DiligentGraphics/DiligentEngine#320] [Q&A]
After replace my own skybox render code with Diligent's `EnvMapRenderer`, 
I found that the Sky is up-side down.

Looking at `ShaderUtilities.fxh`, i found:
```hlsl
float2 TransformDirectionToSphereMapUV(float3 Direction)
{
    float OneOverPi = 0.3183098862;
    return OneOverPi * float2(0.5 * atan2(Direction.z, Direction.x), asin(Direction.y)) + float2(0.5, 0.5);
}
```
It seems that the origin is bottom-left, not the widely used top-left UV origin.

And the sample `Tutorial27_PostProcessing`, flip Y-axis, before sample the environment map.

Weird, is it doing this for compatibility with some OpenGL cubemap generation code?

**Comment by TheMostDiligent** [comment]
There are different conventions, this transform works OK with Diligent applications.
